fe/fe-core/src/main/java/org/apache/doris/analysis/DeleteStmt.java

```diff
@@ -157,7 +157,7 @@ private void constructInsertStmt() throws AnalysisException {
                 expr = new BoolLiteral(true);
             } else if (column.isKey()) {
                 expr = new SlotRef(targetTableRef.getAliasAsName(), column.getName());
-            } else if (!isMow && !column.isVisible()) {
+            } else if (!isMow && (!column.isVisible() || (!column.isAllowNull() && !column.hasDefaultValue()))) {
```

Comment:
please add some comment to explain these conditions

Reply:
This condition is accidentally deleted in https://github.com/apache/doris/pull/26776 due to not considering mor table.